### ISSUE TITLE
Consul token not being exposed to the task with workload identity

### ISSUE THREAD

**Issue by livioribeiro** [hashicorp/nomad#19488]
### Nomad version
1.7.2

### Operating system and Environment details
Ubuntu 22.04 running on a virtual machine

### Issue
As stated in [the nomad documentation](https://developer.hashicorp.com/nomad/docs/job-specification/consul#access-to-token) the consul token would be available to the task via `CONSUL_TOKEN` environment variable and the `/secrets/consul_token` file, however those are not injected into the task. Access to Consul KV works with no problems in template blocks, and Vault token is also injected into the task with no problems

### Reproduction steps
Having a Consul cluster with acl enabled and a Nomad cluster with workload identities configured with Consul integration, run a job using consul as stated in the documentation linked above.

#### Expected Result
Environment variable `CONSUL_TOKEN` defined and file `/secrets/consul_token` created

#### Actual Result
Neither `CONSUL_TOKEN` environment variable nor `/secrets/consul_token` file were present

### Job file (if appropriate)
```hcl
job "example" {

  group "echo" {
    network {
      port "echo" {
        to = 5678
      }
    }

    task "echo" {
      driver = "docker"

      config {
        image = "hashicorp/http-echo"
        ports = ["echo"]
        args  = ["-text", "hello world"]
      }

      identity {
        name = "consul_default"
        aud = ["consul.io"]
      }

      consul {}
      
      template {
        data        = <<-EOT
          CONSUL_TOKEN={{env "CONSUL_TOKEN"}}
        EOT
        destination = "local/config.txt"
      }

      resources {
        cpu    = 100
        memory = 64
      }
    }
  }
}
```

**Comment by pkazmierczak** [comment]
Hey @livioribeiro, thanks for reporting this. I was able to reproduce, and indeed there seems to be a bug in the way we handle Consul tokens for tasks. I'll get right on this.

**Comment by pkazmierczak** [comment]
@livioribeiro thanks again for the bug report, the patch is merged and will be released in 1.7.3.
